cloudsim-plus/src/main/java/org/cloudbus/cloudsim/hosts/HostSimple.java

```diff
@@ -525,6 +522,7 @@ public final Host setActive(final boolean activate) {
 
         this.active = activate;
         notifyStartupOrShutdown(activate, wasActive);
+        datacenter.schedule( 10, CloudSimTags.HOST_POWER_ON_OFF);
```

Comment:
You should introduce new attributes for Host:

- double startupDelay
- double shutDownDelay

Default value is always 0 for numeric attributes. This way, if the researcher doesn't set them, it won't affect current behavior.